src/url_pattern.cpp

```diff
@@ -261,13 +261,13 @@ tl::expected<url_pattern_init, errors> url_pattern_init::process(
 tl::expected<std::string, errors> url_pattern_init::process_protocol(
     std::string_view value, process_type type) {
   ada_log("process_protocol=", value, " [", type, "]");
-  // Let strippedValue be the given value with a single trailing U+003A (:)
-  // removed, if any.
-  if (value.ends_with(":")) {
-    value.remove_suffix(1);
-  }
   // If type is "pattern" then return strippedValue.
   if (type == process_type::pattern) {
+    // Let strippedValue be the given value with a single trailing U+003A (:)
+    // removed, if any.
+    if (value.ends_with(":")) {
+      value.remove_suffix(1);
+    }
     return std::string(value);
   }
   // Return the result of running canonicalize a protocol given strippedValue.
```

Comment:
Maybe we could add as a comment that `canonicalize_protocol` will already remove the trailing `:` if any ? (optional)